deepinterpolation/inferrence_collection.py

```diff
@@ -154,124 +261,190 @@ def __init__(self, inferrence_json_path, generator_obj):
+                              corrected_data: np.ndarray,
+                              corrected_raw: np.ndarray = None,
+                              ):
+        """Write denoised and raw outputs onto respective datasets on
+        h5 file
+        """
+        with h5py.File(self.output_file, "a") as file_handle:
+            dset_out = file_handle[self.output_dataset_name]
+            dset_out[start:end] = np.squeeze(corrected_data, -1)
 
-                predictions_data = self.model.predict(
-                        self.generator_obj,
-                        steps = local_length, max_queue_size = 10,
-                        workers = self.workers,
-                        use_multiprocessing = self.use_multiprocessing)
+            if self.save_raw:
+                raw_out = file_handle[self.raw_dataset_name]
+                raw_out[start:end] = np.squeeze(corrected_raw, -1)
 
-                local_mean, local_std = \
-                        self.generator_obj.__get_norm_parameters__(index_dataset)                
-                
-                local_size = predictions_data.shape[0]
 
+    def run(self):
+        self.model = _load_model(self.json_data)
+        # Initialize onset of output sample movie
+        local_start = self.first_sample
+
+        for epoch_index, index_dataset in enumerate(tqdm(np.arange(self.nb_datasets))):
+            local_data = self.generator_obj[index_dataset]
+            # We overwrite epoch_index to allow the last unfilled epoch
+            self.generator_obj.epoch_index = epoch_index
+            local_mean, local_std = \
+                    self.generator_obj.__get_norm_parameters__(index_dataset)  
+            predictions_data = self.model.predict_on_batch(local_data[0])
+
+            if self.rescale:
+                corrected_data = predictions_data * local_std + local_mean
+            else:
+                corrected_data = predictions_data
+
+            if self.save_raw:
                 if self.rescale:
-                    corrected_data = predictions_data * local_std + local_mean
+                    corrected_raw = local_data[1] * local_std + local_mean
                 else:
-                    corrected_data = predictions_data
-                
-                start = local_start
-                end = local_start + local_size - 1
-
-                # We adjust for the next for loop run
-                local_start = end + 1
-
-                # We squeeze to remove the feature dimension from tensorflow
-                dset_out[start:end+1, :] = np.squeeze(corrected_data, -1)
\ No newline at end of file
+                    corrected_raw = local_data[1]
+
+            local_size = predictions_data.shape[0]
+            start = local_start
+            end = local_start + local_size - 1
+            # We adjust for the next for loop run
+            local_start = end + 1
+            start = self.first_sample + index_dataset * self.batch_size
```

Comment:
`start`, `end` is local to `_write_output_to_file`. Please move this calculation there rather than passing it in. It is currently repeated 3 times.